chain/chain/src/chain.rs

```diff
@@ -931,17 +937,18 @@ impl Chain {
             .into());
         }
 
-        let block_header = self.get_header_by_height(chunk.header.height_included)?.clone();
+        let ShardStateSyncResponse {
+            chunk,
+            chunk_proof,
+            prev_chunk_header,
+            prev_chunk_proof,
+            prev_payload,
+            incoming_receipts_proofs,
+            root_proofs,
+        } = shard_state;
 
         // 1. Checking that chunk header is at least valid
-        // 1a. Checking chunk.header.hash
-        if chunk.header.hash != ChunkHash(hash(&chunk.header.inner.try_to_vec()?)) {
-            byzantine_assert!(false);
-            return Err(ErrorKind::Other(
-                "set_shard_state failed: chunk header hash is broken".into(),
-            )
-            .into());
-        }
+        // 1a. Checking chunk.header.hash - don't check here, hash is computed locally
```

Comment:
let's move this logic into separate function.
I did something like this here - https://github.com/nearprotocol/nearcore/pull/1372/files#diff-5dd0a2f97730038e1709790b3c764bcdR424 so if you do the same it will be easier to merge.

Reply:
But I can also merge this afterwards.

Reply:
That's fine, I moved the logic into check_chunk_validity function